pandas/tests/dtypes/cast/test_promote.py

```diff
@@ -134,7 +134,7 @@ def _check_promote(
         # box_dtype; the expected value returned from maybe_promote is the
         # missing value marker for the returned dtype.
         fill_array = np.array([fill_value], dtype=box_dtype)
-        result_dtype, result_fill_value = maybe_promote(dtype, fill_array)
+        result_dtype, result_fill_value = maybe_promote_with_array(dtype, fill_array)
```

Comment:
@jreback @jbrockmendel 
This is the point which diverts the testing of all array-paths in this module to `maybe_promote_with_array`.